packages/flutter/test/rendering/paragraph_test.dart

```diff
@@ -375,6 +375,41 @@ void main() {
     expect(paragraph.size.height, 30.0);
   }, skip: isBrowser); // https://github.com/flutter/flutter/issues/61018
 
+  group('didExceedMaxLines', () {
+    RenderParagraph createRenderParagraph({
+      int? maxLines,
+      TextOverflow overflow = TextOverflow.clip,
+    }) {
+      return RenderParagraph(
+        const TextSpan(
+          text: 'Here is a long text, maybe exceed maxlines',
+          style: TextStyle(fontSize: 10.0),
+        ),
+        textDirection: TextDirection.ltr,
+        overflow: overflow,
+        maxLines: maxLines,
+      );
+    }
+
+    test('none limited', () {
+      final RenderParagraph paragraph = createRenderParagraph();
+      layout(paragraph, constraints: const BoxConstraints(maxWidth: 100.0));
+      expect(paragraph.didExceedMaxLines, false);
+    });
+
+    test('limited by maxLines', () {
+      final RenderParagraph paragraph = createRenderParagraph(maxLines: 1);
+      layout(paragraph, constraints: const BoxConstraints(maxWidth: 100.0));
+      expect(paragraph.didExceedMaxLines, true);
+    });
+
+    test('limited by ellipsis', () {
+      final RenderParagraph paragraph = createRenderParagraph(overflow: TextOverflow.ellipsis);
+      layout(paragraph, constraints: const BoxConstraints(maxWidth: 100.0));
+      expect(paragraph.didExceedMaxLines, true);
```

Comment:
Is this right? `maxLines` is null in this test, no?

Reply:
Due to this, [`TextPainter.didExceedMaxLines`](https://github.com/flutter/flutter/blob/c02ace6b4886810993542b8871d45d57d5b81d82/packages/flutter/lib/src/painting/text_painter.dart#L1070)

> If [maxLines] is null, this is true if [ellipsis] is not the empty string and there was a line that overflowed the `maxWidth` argument passed to [layout]; otherwise it is false.

So that's right.

In previous commit, i add some comment for this. Then inspired by @AlexV525 , i think it's `TextPainter.didExceedMaxLines`'s behavior, maybe doesn't need to be repeated here.